internal/metamorph/store/postgresql/postgres.go

```diff
@@ -464,12 +464,38 @@ func (p *PostgreSQL) UpdateStatusBulk(ctx context.Context, updates []store.Updat
 		;
     `
 
-	rows, err := p.db.QueryContext(ctx, qBulk, pq.Array(txHashes), pq.Array(statuses), pq.Array(rejectReasons), p.hostname, metamorph_api.Status_SEEN_ON_NETWORK, metamorph_api.Status_MINED)
+	tx, err := p.db.Begin()
 	if err != nil {
+		panic(err)
+	}
+
+	_, err = tx.Exec("LOCK TABLE metamorph.transactions IN EXCLUSIVE MODE")
+	if err != nil {
+		if err := tx.Rollback(); err != nil {
+			panic(err)
```

Comment:
panic stops the whole program from execution. I think we should simply return an error here.